client/http_sender.go

```diff
@@ -56,25 +47,23 @@ func init() {
 
 // httpSender is an implementation of Sender which exposes the
 // Key-Value database provided by a Cockroach cluster by connecting
-// via HTTP to a Cockroach node. Overly-busy nodes will redirect
-// this client to other nodes.
+// via HTTP to a Cockroach node.
 type httpSender struct {
-	server    string        // The host:port address of the Cockroach gateway node
-	client    *http.Client  // The HTTP client
-	context   *base.Context // The base context: needed for client setup.
-	retryOpts retry.Options
+	ctx PostContext
 }
 
 // newHTTPSender returns a new instance of httpSender.
 func newHTTPSender(server string, ctx *base.Context, retryOpts retry.Options) (*httpSender, error) {
 	sender := &httpSender{
-		server:    server,
-		context:   ctx,
-		retryOpts: retryOpts,
+		ctx: PostContext{
+			Server:    server,
+			Endpoint:  KVDBEndpoint,
+			Context:   ctx,
+			RetryOpts: retryOpts,
+		},
 	}
-	var err error
-	sender.client, err = ctx.GetHTTPClient()
-	if err != nil {
+	// Ensure that the context returns a HTTPClient
+	if _, err := ctx.GetHTTPClient(); err != nil {
```

Comment:
ah ha client.TestOpenArgs(t *testing.T) is failing because it depends on a non lazy creation of the HTTPClient.